plugin/src/main/java/org/ehrbase/plugin/security/PluginSecurityConfiguration.java

```diff
@@ -0,0 +1,156 @@
+    public void set$AroundClosure(AroundClosure arc) { throw new UnsupportedOperationException(); }
+    public Object proceed() throws Throwable { return invocation.proceed(); }
+
+    public Object proceed(Object[] args) throws Throwable {
+      Method method = invocation.getMethod();
+      Object obj = invocation.getThis();
+      return method.invoke(obj, args);
+    }
+  }
+  
+  private abstract static class AspectAdapter implements MethodInterceptor {
+    private final AnnotationAspect aspect;
+
+    AspectAdapter(AnnotationAspect aspect) { this.aspect = aspect; }
+    protected AnnotationAspect getAspect() { return aspect; }
+    public abstract Object invoke(MethodInvocation invocation) throws Throwable;
+  }
+
+  public static class AnyOfAspectsCondition extends AnyNestedCondition {
+    public AnyOfAspectsCondition() {
+      super(ConfigurationPhase.REGISTER_BEAN);
+    }
+    
+    @ConditionalOnBean(value = AuthorizationAspect.class)
+    static class OnAuthorization {} 
+    
+    @ConditionalOnBean(value = TenantAspect.class)
+    static class OnTenant {}
+  }
+  
+  private ApplicationContext applicationContext;
+  
+  @Bean
+  @Conditional(AnyOfAspectsCondition.class)
+  public DefaultAdvisorAutoProxyCreator advisorAutoProxyCreator() {
+    return new DefaultAdvisorAutoProxyCreator();
+  }
+  
+  @Bean
+  @ConditionalOnBean(value = AuthorizationAspect.class)
+  public Advisor authorizationAspect() {
+    ApplicationContext parentCtx = applicationContext.getParent();
+    AuthorizationAspect theAspect = parentCtx.getBean(AuthorizationAspect.class);
+
+    return new DefaultPointcutAdvisor(
+        new AnnotationMatchingPointcut(null, EhrbaseAuthorization.class, true),
+        new AspectAdapter(theAspect) {
+          public Object invoke(MethodInvocation invocation) throws Throwable {
+            return getAspect().action(new ProceedingJoinPointAdapter(invocation), null);
+          }
+        });
+  }
+  
+  @Bean
+  @ConditionalOnBean(value = TenantAspect.class)
+  public Advisor tenantAspect() {
+    ApplicationContext parentCtx = applicationContext.getParent();
+    TenantAspect theAspect = parentCtx.getBean(TenantAspect.class);
+
+    return new DefaultPointcutAdvisor(
+        new AnnotationMatchingPointcut(TenantAware.class, true),
+        new AspectAdapter(theAspect) {
+          public Object invoke(MethodInvocation invocation) throws Throwable {
+            Method method = invocation.getMethod();
+            Class<?> declaringClass = method.getDeclaringClass();
+            TenantAware annotation =  Objects.requireNonNull(declaringClass.getAnnotation(TenantAware.class));
+            return getAspect().action(new ProceedingJoinPointAdapter(invocation), List.of(annotation));
+          }
+        });
+  }
+
+  public void setApplicationContext(ApplicationContext applicationContext) throws BeansException {
+    this.applicationContext = applicationContext;
+  }
+  // @format:on
```

Comment:
please move this to Line 90. For the adapter definitions its fine, but it should not exclude the entire class from formatting. Also please make sure that at least the indentation matches.

Reply:
Done